recipes/daw_header_libraries/all/conanfile.py

```diff
@@ -1,71 +1,70 @@
-import os
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.files import get, copy
+from conan.tools.build import check_min_cppstd
+from conan.tools.scm import Version
+from conan.tools.layout import basic_layout
 
-from conans.errors import ConanInvalidConfiguration
-from conans import ConanFile, CMake, tools
+import os
 
-required_conan_version = ">=1.43.0"
+required_conan_version = ">=1.51.3"
 
 class DawHeaderLibrariesConan(ConanFile):
     name = "daw_header_libraries"
+    description = "Various header libraries mostly future std lib, replacements for(e.g. visit), or some misc"
     license = "BSL-1.0"
     url = "https://github.com/conan-io/conan-center-index"
-    description = "Various header libraries mostly future std lib, replacements for(e.g. visit), or some misc"
-    topics = ("algorithms", "helpers", "data-structures")
     homepage = "https://github.com/beached/header_libraries"
-    settings = "compiler",
-    generators = "cmake",
+    topics = ("algorithms", "helpers", "data-structures")
+    settings = "os", "arch", "compiler", "build_type"
     no_copy_source = True
 
-    _compiler_required_cpp17 = {
-        "Visual Studio": "16",
-        "gcc": "8",
-        "clang": "7",
-        "apple-clang": "12.0",
-    }
-
     @property
-    def _source_subfolder(self):
-        return "source_subfolder"
+    def _compiler_required_cpp17(self):
+        return {
+            "Visual Studio": "16",
+            "msvc": "14.4",
+            "gcc": "8",
+            "clang": "7",
+            "apple-clang": "12.0",
+        }
+
+    def layout(self):
+        basic_layout(self, src_folder="src")
+
+    def package_id(self):
+        self.info.clear()
 
     def validate(self):
         if self.settings.get_safe("compiler.cppstd"):
-            tools.check_min_cppstd(self, "17")
+            check_min_cppstd(self, "17")
 
-        minimum_version = self._compiler_required_cpp17.get(str(self.settings.compiler), False)
-        if minimum_version:
-            if tools.Version(self.settings.compiler.version) < minimum_version:
-                raise ConanInvalidConfiguration("{} requires C++17, which your compiler does not support.".format(self.name))
+        minimum_version = self._compiler_required_cpp17.get(str(self.info.settings.compiler), False)
+        if minimum_version and Version(self.info.settings.get_safe("compiler.version")) < minimum_version:
+            raise ConanInvalidConfiguration("{} requires C++17, which your compiler does not support.".format(self.name))
```

Comment:
self.info in validate() can't work yet in a conan v1 client

Reply:
@SpaceIm 
Sorry, I don't know how to modify it, could you please tell me?

I wrote a recipe referring to the following template.
https://github.com/conan-io/conan-center-index/blob/master/docs/package_templates/header_only/all/conanfile.py#L61-L64

Do we need to change the template?

Reply:
@SpaceIm 
Now I understood by https://github.com/conan-io/conan-center-index/pull/12928#discussion_r971033689.
Thanks.